### PR TITLE
return 0 on true, -1 on false, other negative on error

### PR DESCRIPTION
**Pull request by milyin**
API update accordingly to variant (1) proposed in this discussion https://github.com/eclipse-zenoh/zenoh-pico/issues/173

### CODE REVIEW

**Comment by p-avital** [comment]
This would break any code that relies on `z_keyexpr_intersect(a, b) >= 0` to assert intersection. Do we take that risk? @cguimaraes

**Comment by cguimaraes** [comment]
What worries me is that we might break the logic of anyone already making use of them. So, we will need to be clear on the change log and documentation.

But at the same time, I also agree that the functions below should return the same type:
```
int8_t z_keyexpr_intersects(z_keyexpr_t l, z_keyexpr_t r);
_Bool zp_keyexpr_intersect_null_terminated(const char *l, const char *r);
```

In between, Zenoh-Pico and Zenoh-C what does provide less impact for the user?

File: src/api/api.c [hunk 1]
```diff
@@ -94,48 +94,77 @@ int8_t zp_keyexpr_canonize_null_terminated(char *start) {
...
+    return ret;
 }
 
-_Bool zp_keyexpr_intersect_null_terminated(const char *l, const char *r) {
-    return _z_keyexpr_intersects(l, strlen(l), r, strlen(r));
+int8_t zp_keyexpr_intersect_null_terminated(const char *l, const char *r) {
+    int8_t ret = -1;
+
+    if (l != NULL && r != NULL) {
+        ret = (_z_keyexpr_intersects(l, strlen(l), r, strlen(r)) == true) ? 0 : -1;
```
Comment (p-avital) [inline src/api/api.c:136]: why use `== true` for a ternary? MISRA?

**Review by p-avital — APPROVED** [review]

File: src/api/api.c (hunk 1 above)
Comment (cguimaraes) [reply src/api/api.c:136]: Indeed, the explicitly comparison was introduced on the codebase with the MISRA compliance.

**Review by cguimaraes — APPROVED** [review]

**Pull request merged**